longest_palindrome

Farhan-labib opened the pull request:
### Describe your change:



* [✔️ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [✔️] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [✔️ ] This pull request is all my own work -- I have not plagiarized.
* [✔️ ] I know that pull requests will not be merged if they fail the automated tests.
* [✔️ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [✔️ ] All new Python files are placed inside an existing directory.
* [ ✔️] All filenames are in all lowercase characters with no spaces or dashes.
* [ ✔️] All functions and variable names follow Python naming conventions.
* [ ✔️] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [✔️ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

imSanko reviewed (changes requested):
Nice but can you add tests for more enhancement.

Farhan-labib commented:
@imSanko  

Enhanced Test Cases: Added additional test cases, including complex and large strings, to thoroughly evaluate the function's performance and accuracy.

Benchmarking: Incorporated benchmarking to measure the execution time for each test case. This helps in understanding the performance of the function with different inputs.
